### ISSUE TITLE
New Setting: Automatically switch to orthographic mode when using sideview

### ISSUE THREAD

**Issue by Minecreator200** [SoftFever/OrcaSlicer#7848]
### Is there an existing issue for this feature request?

- [x] I have searched the existing issues

### Is your feature request related to a problem?

I wouldn't say it's a problem, but think that it would be a **great enhancement in user friendliness**.
It would make placing objects way easier as opposed to perspective view.

### Which printers will be beneficial to this feature?

All

### Describe the solution you'd like

Whenever the setting is active and all camera angles are a multiple of 90deg switch to orthographic mode.
As already stated above, it would make placing objects way easier.

Pseudo-Code:
```
if (automaticChangeEnabled)
    Renderer.setCameraMode(
        (camera.rotX % 90 == 0 && camera.rotY % 90 == 0 && camera.rotZ % 90 == 0) ?
        CameraMode.ORTHOGRAPHIC :
        CameraMode.PERSPECTIVE
    );
```

**Comment by buzzhuzz** [comment]
Having this feature implemented as proposed may result in jerky view since projection might be changed while orbiting build plate.

Better implementation might be "automatically switch to orthographic projection on setting camera view using Ctrl+<1..6>. Switch back to perspective on camera rotation"

**Comment by Minecreator200** [comment]
Couldn't we just make it so it only does it if the small cube in the bottom left corner of the view is clicked? That's probably a good compromise.

**Comment by Noisyfox** [comment]
> Couldn't we just make it so it only does it if the small cube in the bottom left corner of the view is clicked? That's probably a good compromise.

I remember saw similar behavior in some 3D modeling softwares.

**Comment by Minecreator200** [comment]
> > Couldn't we just make it so it only does it if the small cube in the bottom left corner of the view is clicked? That's probably a good compromise.
> 
> I remember saw similar behavior in some 3D modeling softwares, though I can't tell exactly which one.

Yes, Blender does have this feature for example. Some like it, some don't. That's why I had the idea of the toggle switch in the settings.

**Comment by Noisyfox** [comment]
Please try https://github.com/SoftFever/OrcaSlicer/pull/8312

**Comment by Minecreator200** [comment]
See Noisyfoxes pull request #8312
